app/recordtransfer/views/profile.py

```diff
@@ -106,6 +114,34 @@ def form_invalid(self, form: BaseModelForm) -> HttpResponse:
         return super().form_invalid(profile_form)
 
 
+@require_http_methods(["GET", "DELETE"])
+def in_progress_submission(request: HttpRequest, uuid: str) -> HttpResponse:
+    """Handle GET (show modal) and DELETE (delete submission) for in-progress submissions. Both
+    requests must be made by HTMX, or else a 400 Error is returned.
+    """
+    if not request.htmx:
+        return HttpResponse(status=400)
+
+    if request.method == "GET":
+        in_progress = get_object_or_404(InProgressSubmission, uuid=uuid, user=request.user)
+        context = {"in_progress": in_progress}
+        return render(request, "includes/delete_in_progress_submission_modal.html", context)
+
+    # DELETE request
+    try:
+        in_progress = InProgressSubmission.objects.get(uuid=uuid, user=request.user)
+        in_progress.delete()
+        response = HttpResponse(status=204)
+        return trigger_client_event(
+            response, "showSuccess", {"value": "In-progress submission deleted."}
+        )
+    except Exception:
+        response = HttpResponse(status=500)
+        return trigger_client_event(
+            response, "showError", {"value": "Failed to delete in-progress submission."}
+        )
```

Comment:
I think get_object_or_404 should be used here, too. You'll have to change the Exception to let those exceptions through because that function raises an `Http404` exception. If you leave it as `except Exception` you'll mask the `Http404` exception as a 500 error